core/dbt/context/providers.py

```diff
@@ -684,8 +684,15 @@ def resolve(self, source_name: str, table_name: str):
                 target_kind="source",
                 disabled=(isinstance(target_source, Disabled)),
             )
+
+        # Source quoting does _not_ respect global configs in dbt_project.yml, as documented here:
+        # https://docs.getdbt.com/reference/project-configs/quoting
+        # Use an object with an empty quoting field to bypass any settings in self.
+        class SourceQuotingConfig:
+            quoting: Dict[str, Any] = {}
+
         return self.Relation.create_from(
-            self.config,
+            SourceQuotingConfig(),
```

Comment:
An alternative approach would be to extend `BaseRelation.create_from` to accept an `Optional[HasQuoting]` value instead of just `HasQuoting`. That said, the method in dbt-adapters is:
* abstracted to the point where it does not concern itself with the node type (e.g. model vs seed vs source, etc), and this kind of business logic feels better-suited in dbt-core
* overridable across adapters, so could lead to inconsistent handling + need updates across multiple adapters for consistent behaviour

So I went with this approach. Open to feedback :) I opted to define `SourceBaseQuotingConfig` inline here since it quite specialized and likely won't need to be used elsewhere in the codebase.